### PR TITLE
Add git hooks with husky and github action for linting

### PR DESCRIPTION
**Pull request by coreyar**
- Rationalizes prettier and eslint
- Adds precommit hooks for linting and formatting
- Adds github action for linting and formatting
- Runs updated config on all files

### CODE REVIEW

**Review by baomastr — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -7,8 +7,9 @@
     "build": "react-scripts --max_old_space_size=4096 build",
     "test": "react-scripts test",
     "eject": "react-scripts eject",
+    "pretty": "yarn prettier src/**/** public/**/*.html --check",
     "lint": "export NODE_OPTIONS=\"--max-old-space-size=4096\" && eslint .",
-    "lint:fix": "eslint . --fix",
+    "lint:fix": "yarn lint --fix && yarn pretty --write",
```
Comment (baomastr) [inline package.json:12]: what do you think about this method for local development?
https://gist.github.com/estorgio/e8bcaa8e87d0fcdcf85fdf598956e34c

Comment (coreyar) [reply package.json:12]: I am not a fan of automatically adding files. A developer may have edited files that don't belong in the commit. Is there anything else you see we are missing from that setup?

Comment (baomastr) [reply package.json:12]: this is needed to automatically comb the code in the files that were modified according to our codestyle

Comment (coreyar) [reply package.json:12]: The precommit hook currently doesn't modify files, it only checks that they pass linting/ formatting and aborts the commit if there is an error

**Pull request merged**